ui/src/components/ImportDialog.tsx

```diff
@@ -1,131 +1,238 @@
+      await importVolume({
+        volumeName: selectedVolumeName || volumeName,
+        path,
+      });
+      onClose(true);
+    } else {
+      await loadImage({
+        volumeName: selectedVolumeName || volumeName,
+        imageName: image,
+      });
+      onClose(true);
+    }
+  };
 
-        ddClient.extension.vm.service
-            .get(`/volumes/${context.store.volumeName}/import?path=${path}`)
-            .then((_: any) => {
-                actionSuccessfullyCompleted = true
-                ddClient.desktopUI.toast.success(
-                    `File ${path} imported into volume ${context.store.volumeName}`
-                );
-            })
-            .catch((error) => {
-                actionSuccessfullyCompleted = false
-                ddClient.desktopUI.toast.error(
-                    `Failed to import file ${path} into volume ${context.store.volumeName}: ${error.message}. HTTP status code: ${error.statusCode}`
-                );
-            })
-            .finally(() => {
-                setActionInProgress(false);
-                setPath("");
-                props.onClose(actionSuccessfullyCompleted)
-            })
-    };
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setFromRadioValue(
+      (event.target as HTMLInputElement).value as "file" | "image"
+    );
+  };
 
+  const renderFormControlFile = () => {
     return (
-        <Dialog open={props.open} onClose={props.onClose}>
-            <DialogTitle>Import gzip'ed tarball into a volume</DialogTitle>
-            <DialogContent>
-                <Backdrop
-                    sx={{
-                        backgroundColor: "rgba(245,244,244,0.4)",
-                        zIndex: (theme) => theme.zIndex.drawer + 1,
-                    }}
-                    open={actionInProgress}
-                >
-                    <CircularProgress color="info"/>
-                </Backdrop>
-                <DialogContentText>
-                    Extracts a gzip'ed tarball into a volume.
-                </DialogContentText>
-
-                <Grid container direction="column" spacing={2}>
-                    <Grid item>
-                        <Typography variant="body1" color="text.secondary" sx={{mt: 2}}>
-                            Choose a .tar.gz to import, e.g. file.tar.gz
-                        </Typography>
-                    </Grid>
-                    <Grid item>
-                        <Button variant="contained" onClick={selectImportTarGzFile}>
-                            Select file
-                        </Button>
-                    </Grid>
+      <>
+        <FormControlLabel value="file" control={<Radio />} label="Local file" />
+        {fromRadioValue === "file" && (
+          <Stack pt={1} pb={2} pl={4}>
+            <Typography pb={1} variant="body2">
+              Select a file (.tar.gz) whose content is to be imported into the
+              new volume.
```

Comment:
When `renderFormControlFile` is rendered when importing from an existing volume, the word `new` is confusing.

When using the `Import into new volume` button, use the following copy:

`Select a file (.tar.gz) whose content is to be imported into the new volume.`

When using clicking on the import button of an existing volume, use the following copy:

`Select a file (.tar.gz) whose content is to be imported into the existing volume.`

Reply:
Totally right, I missed that line. Fixed!